libminifi/src/FlowController.cpp

```diff
@@ -163,6 +174,31 @@ FlowController::~FlowController() {
   provenance_repo_ = nullptr;
 }
 
+bool FlowController::applyConfiguration(std::string &configurePayload) {
```

Comment:
Right now we're beginning to tightly couple "listening" to flow controller. We can improve that by moving to have a package that isolates much of the C2 specifics. With a call back, you might be able to give call flowconfiguration, which is part of flow controller and it doesn't need to know the state.